### PR TITLE
[tools/resourcedocsgen] Generate a package tree along with the resource docs

### PR DESCRIPTION
**Pull request by praneetloke**
### Proposed changes

Just minimal changes here to absorb the changes from https://github.com/pulumi/pulumi/pull/8102.

### Unreleased product version (optional)

N/A

### Related issues (optional)

https://github.com/pulumi/pulumi-service/issues/7346

### CODE REVIEW

File: tools/resourcedocsgen/cmd/metadata.go [hunk 1]
```diff
@@ -73,6 +73,73 @@ var categoryLookup = map[string]pkg.PackageCategory{
 	"sumologic":     pkg.PackageCategoryMonitoring,
 }
 
+// TODO[pulumi/pulumi#7813]: Remove this lookup once display name is available in
+// the Pulumi schema.
+// titleLookup is a map to map module package name to the desired display name
+// for display in the TOC menu under API Reference.
+// NOTE: For the time being this lookup map and the one used by the docs
+// generator in `pulumi/pulumi` must be kept up-to-date.
+var titleLookup = map[string]string{
```
Comment (praneetloke) [inline tools/resourcedocsgen/cmd/metadata.go:82]: I am sorry for this but it's needed without need to make more changes in the docs generator in `pulumi/pulumi` to use the existing title lookup here. In any case, both will be removed when pulumi/pulumi#7813 is implemented.

Comment (praneetloke) [reply tools/resourcedocsgen/cmd/metadata.go:82]: @infin8x fyi.

Comment (infin8x) [reply tools/resourcedocsgen/cmd/metadata.go:82]: Ah nice, thanks! Once this PR is in, can I propose some changes to these titles? (Or can I just do it here? Don't want to get in the way at all)

Comment (praneetloke) [reply tools/resourcedocsgen/cmd/metadata.go:82]: Absolutely, please feel free to do it here, either as a comment or as a GH suggestion against the appropriate lines and I'll accept them.

File: tools/resourcedocsgen/cmd/metadata.go [hunk 2]
```diff
@@ -73,6 +73,73 @@ var categoryLookup = map[string]pkg.PackageCategory{
...
+	"aiven":         "Aiven",
+	"akamai":        "Akamai",
+	"alicloud":      "AliCloud",
```
Comment (infin8x) [inline tools/resourcedocsgen/cmd/metadata.go:85]:
```suggestion
	"alicloud":      "Alibaba Cloud",
```

File: tools/resourcedocsgen/cmd/metadata.go [hunk 3]
```diff
@@ -73,6 +73,73 @@ var categoryLookup = map[string]pkg.PackageCategory{
...
+	"auth0":         "Auth0",
+	"aws":           "AWS",
+	"azure":         "Azure Classic",
+	"azure-native":  "Azure Native",
+	"azuread":       "Azure AD",
```
Comment (infin8x) [inline tools/resourcedocsgen/cmd/metadata.go:90]:
```suggestion
	"azuread":       "Azure Active Directory",
```

File: tools/resourcedocsgen/cmd/metadata.go [hunk 4]
```diff
@@ -73,6 +73,73 @@ var categoryLookup = map[string]pkg.PackageCategory{
...
+	"kong":          "Kong",
+	"kubernetes":    "Kubernetes",
+	"linode":        "Linode",
+	"mailgun":       "Mailgun",
+	"mongodbatlas":  "MongoDB Atlas",
+	"mysql":         "MySQL",
+	"newrelic":      "New Relic",
+	"ns1":           "NS1",
+	"okta":          "Okta",
+	"openstack":     "Open Stack",
```
Comment (infin8x) [inline tools/resourcedocsgen/cmd/metadata.go:121]:
```suggestion
	"openstack":     "OpenStack",
```

**Review by infin8x — APPROVED** [review]

File: tools/resourcedocsgen/cmd/metadata.go [hunk 5]
```diff
@@ -73,6 +73,73 @@ var categoryLookup = map[string]pkg.PackageCategory{
...
+	"cloudinit":     "cloud-init",
+	"consul":        "Consul",
+	"datadog":       "Datadog",
+	"digitalocean":  "DigitalOcean",
+	"dnsimple":      "DNSimple",
+	"docker":        "Docker",
+	"eks":           "EKS",
+	"f5bigip":       "f5 BIG-IP",
+	"fastly":        "Fastly",
+	"gcp":           "GCP",
```
Comment (infin8x) [inline tools/resourcedocsgen/cmd/metadata.go:105]:
```suggestion
	"gcp":           "Google Cloud Classic",
```

File: tools/resourcedocsgen/cmd/metadata.go [hunk 6]
```diff
@@ -73,6 +73,73 @@ var categoryLookup = map[string]pkg.PackageCategory{
...
+	"google-native": "Google Native (preview)",
```
Comment (infin8x) [inline tools/resourcedocsgen/cmd/metadata.go:106]:
```suggestion
	"google-native": "Google Cloud Native (preview)",
```

File: tools/resourcedocsgen/cmd/metadata.go (hunk 1 above)
Comment (infin8x) [reply tools/resourcedocsgen/cmd/metadata.go:82]: Perfect, done.

**Review by justinvp — APPROVED** [review]

**Comment by praneetloke** [comment]
Ported over the title lookup updates to the docs generator in `pulumi/pulumi`. I also pulled in the change from there that was done yesterday https://github.com/pulumi/pulumi/pull/8101. cc @infin8x

**Comment by infin8x** [comment]
SGTM @praneetloke. I'm 👍🏻  whenever this is ready to go

**Pull request merged**